models/classes/class.KeyValueResultStorage.php

```diff
@@ -234,8 +234,9 @@ public function getVariables($callId)
 
         if (is_array($callId)) {
             foreach ($callId as $id) {
-                $variables = array_merge($variables, $this->getVariables($id));
+                $variables[] = $this->getVariables($id);
```

Comment:
If the input arrays have the same string keys, then the later value for that key will overwrite the previous one. 
If, however, the arrays contain numeric keys, the later value will not overwrite the original value, but will be appended.

Values in the input arrays with numeric keys will be renumbered with incrementing keys starting from zero in the result array.
are you sure that we are not changing the scope ?

Reply:
I would avoid this PR, since it is optional. I think this change is a bit risky.